### PR TITLE
PM-10280 - Implement Autofill Setup Completion Screen

### PR DESCRIPTION
**Pull request by phil-livefront**
## 🎟️ Tracking

[PM-10280](https://bitwarden.atlassian.net/browse/PM-10280)

## 📔 Objective

- Updated `ExtensionActivationView` to match new designs when completing autofill set up.  This change is gated behind the `native-create-account-flow` feature flag
- Added a universal link to deep link back to Bitwarden from the extension

## 📸 Screenshots

https://github.com/user-attachments/assets/ce3cc8f6-a793-4114-ba4c-3e464a81910e

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

[PM-10280]: https://bitwarden.atlassian.net/browse/PM-10280?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: BitwardenShared/UI/Platform/ExtensionSetup/ExtensionActivation/ExtensionActivationView.swift [hunk 1]
```diff
@@ -8,87 +8,113 @@ struct ExtensionActivationView: View {
...
-                Localizations.autofillSetup2
+    // MARK: View
+
+    var body: some View {
+        Group {
+            if store.state.showLegacyView {
+                legacyContent
+            } else {
+                content
+            }
```
Comment (matt-livefront) [inline BitwardenShared/UI/Platform/ExtensionSetup/ExtensionActivation/ExtensionActivationView.swift:28]: ⚠️ `legacyContent` handles two cases based on the `extensionType` in the state: 1) activating the autofill extension and 2) activating the action extension. These new changes should only affect activating the autofill extension (1). If you activate the action extension (app Settings > Auto-fill > App extension), you should still see the legacy content for the app extension, but you'll currently see the new changes for the autofill extension. Would it make sense to still show the legacy view for the app extension type?

File: BitwardenShared/Core/Platform/Utilities/ExternalLinksConstants.swift [hunk 2]
```diff
@@ -10,6 +10,9 @@ enum ExternalLinksConstants {
     /// A link to Bitwarden's organizations information webpage.
     static let aboutOrganizations = URL(string: "https://bitwarden.com/help/about-organizations")!
 
+    /// A deep link to the Bitwarden app.
+    static let appDeepLink = URL(string: "https://www.vault.bitwarden.pw/redirect-connector.html#")!
```
[the remark this answers is not in the document]
Comment (phil-livefront) [reply BitwardenShared/Core/Platform/Utilities/ExternalLinksConstants.swift:14]: I was going back and forth on this so good call out.  I like the universal link purely for UX but this doesn't seem like a good link to reply on (i wish we had a standard one to use for this type of case).  I'll use `bitwarden://` for this, and I can bring up adding the ability for us in the future to handle a more generic universal link to open the app.

File: BitwardenShared/UI/Platform/ExtensionSetup/ExtensionActivation/ExtensionActivationView.swift (hunk 1 above)
Comment (phil-livefront) [reply BitwardenShared/UI/Platform/ExtensionSetup/ExtensionActivation/ExtensionActivationView.swift:28]: got this fixed up, great catch.

File: BitwardenShared/UI/Platform/Application/AppProcessor.swift [hunk 3]
```diff
@@ -102,6 +102,8 @@ public class AppProcessor {
     /// - Parameter url: The deep link URL to handle.
     ///
     public func openUrl(_ url: URL) async {
+        guard url.scheme == "otpauth" else { return }
```
Comment (fedemkr) [inline BitwardenShared/UI/Platform/Application/AppProcessor.swift:105]: 🤔 Could this be extracted to some constant or just an URL extension? I see it's also being used [here](https://github.com/bitwarden/ios/blob/main/BitwardenShared/Core/Vault/Services/TOTP/OTPAuthModel.swift#L67).

File: BitwardenShared/UI/Platform/Application/Support/Localizations/en.lproj/Localizable.strings [hunk 4]
```diff
@@ -107,6 +107,8 @@
 "ExtensionInSafari" = "In Safari, find Bitwarden using the share icon (hint: scroll to the right on the bottom row of the menu).";
 "ExtensionInstantAccess" = "Get instant access to your passwords!";
 "ExtensionReady" = "You're ready to log in!";
+"ExtensionAllSet" = "You're all set!";
+"ExtensionAutoFillActivated" = "You can now use autofill to log into apps and websites using your saved passwords. Now, you can explore everything else Bitwarden has to offer.";
```
Comment (fedemkr) [inline BitwardenShared/UI/Platform/Application/Support/Localizations/en.lproj/Localizable.strings:111]: 🎨 First one, update key to match the values; in the second one we can use the `DescriptionLong` suffix so we maintain the same approach as with other long strings.
```suggestion
"YouReAllSet" = "You're all set!";
"ExtensionAutoFillActivatedDescriptionLong" = "You can now use autofill to log into apps and websites using your saved passwords. Now, you can explore everything else Bitwarden has to offer.";
```

Comment (phil-livefront) [reply BitwardenShared/UI/Platform/Application/Support/Localizations/en.lproj/Localizable.strings:111]: sounds good.  I thought I was following the pattern here but I'll make the change you sggested. 

```
"ExtensionReenable" = "Reactivate app extension";
"ExtensionAlmostDone" = "Almost done!";
"ExtensionEnable" = "Activate app extension";
"ExtensionInSafari" = "In Safari, find Bitwarden using the share icon (hint: scroll to the right on the bottom row of the menu).";
"ExtensionInstantAccess" = "Get instant access to your passwords!";
"ExtensionReady" = "You're ready to log in!";
"ExtensionAllSet" = "You're all set!";
"ExtensionAutoFillActivated" = "You can now use autofill to log into apps and websites using your saved passwords. Now, you can explore everything else Bitwarden has to offer.";
"ExtensionSetup" = "Your logins are now easily accessible from Safari, Chrome, and other supported apps.";
"ExtensionSetup2" = "In Safari and Chrome, find Bitwarden using the share icon (hint: scroll to the right on the bottom row of the share menu).";
"ExtensionTapIcon" = "Tap the Bitwarden icon in the menu to launch the extension.";
"ExtensionTurnOn" = "To turn on Bitwarden in Safari and other apps, tap the \"more\" icon on the bottom row of the menu.";
```

File: BitwardenShared/UI/Platform/Application/Support/Localizations/en.lproj/Localizable.strings [hunk 5]
```diff
@@ -101,14 +101,12 @@
 "EditItem" = "Edit item";
 "EnableAutomaticSyncing" = "Allow automatic syncing";
 "EnterEmailForHint" = "Enter your account email address to receive your master password hint.";
-"ExntesionReenable" = "Reactivate app extension";
+"ExtensionReenable" = "Reactivate app extension";
```
Comment (fedemkr) [inline BitwardenShared/UI/Platform/Application/Support/Localizations/en.lproj/Localizable.strings:104]: ⛏️ Given the key is being modified, could we change the it to `ReactivateAppExtension`?

Comment (phil-livefront) [reply BitwardenShared/UI/Platform/Application/Support/Localizations/en.lproj/Localizable.strings:104]: sure!  makes sense.  I just saw the spelling error and figured id fix it but this makes more sense.  Thank you!

**Review by fedemkr — APPROVED** [review]

**Review by matt-livefront — APPROVED** [review]
Looks good!

**Review by matt-livefront — APPROVED** [review]

**Pull request merged**